src/dataframe/dataframe.jl

```diff
@@ -278,6 +278,26 @@ _columns(df::DataFrame) = getfield(df, :columns)
 nrow(df::DataFrame) = ncol(df) > 0 ? length(_columns(df)[1])::Int : 0
 ncol(df::DataFrame) = length(index(df))
 
+##############################################################################
+##
+## DataFrame consistency check
+##
+##############################################################################
+
+function _check_consistency(df::DataFrame)
+    cols, idx = _columns(df), index(df)
+    ncols = length(cols)
+    @assert length(idx.names) == length(idx.lookup) == ncols "Index and columns of a data frame do not match"
```

Comment:
More precisely, lengths don't match. This should never happen, right? If so, maybe just keep the default printing.

Reply:
removed